src/error.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2019 O.S. Systems Sofware LTDA
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::ffi;
+use std::ffi::CStr;
+use thiserror::Error;
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Extraction error: '{0}'")]
+    ExtractionError(String),
+
+    #[error("Nix error: '{0}'")]
+    Nix(#[from] nix::Error),
+
+    #[error("Io error: '{0}'")]
+    Io(#[from] std::io::Error),
+
+    #[error("Utf error: '{0}'")]
+    Utf(#[from] std::str::Utf8Error),
+
+    #[error("Try from int error: '{0}'")]
+    TryInt(#[from] std::num::TryFromIntError),
+
+    #[error("Error to create the archive struct, is null")]
+    ArchiveNull,
+
+    #[error("The entry is null, failed to set the pathname")]
+    EntryNull,
+
+    #[error("File not found")]
+    FileNotFound,
+
+    #[error("The end of file")]
+    EndOfFile,
+}
+
+pub(crate) fn from_archive(value: i32, archive: *mut ffi::archive) -> Result<()> {
```

Comment:
Minor thing, I'd name this as  `archive_ok_or_error` what do you think?

Reply:
How about `archive_result` then? "or_or_error" essentially means "result".

Reply:
or `ensure_archive_ok`.

Reply:
`archive_ok_error` has more meaning than `from_archive` on that context, looks good to me. My only complain about it it`is just a little verbose.